de.metas.adempiere.adempiere/base/src/main/java/de/metas/location/impl/AddressBuilder.java

```diff
@@ -648,15 +655,19 @@ else if ("CR".equals(token))
 	 * @param trxName
 	 * @return
 	 */
-	private String buildUserBlock(final Properties ctx, final boolean isLocal, final I_AD_User user, final String bPartnerBlock, final boolean isPartnerCompany, final String trxName)
+	private String buildUserBlock(@NonNull final org.compiere.model.I_C_BPartner bPartner,  final boolean isLocal, final I_AD_User user, final String bPartnerBlock, final String trxName)
 	{
-		String userGreeting = "";
+		final Properties ctx = InterfaceWrapperHelper.getCtx(bPartner);
+		final boolean isPartnerCompany = bPartner.isCompany();
+		final Language language = Language.asLanguage(bPartner.getAD_Language());
```

Comment:
BP's language is not mandatory....

Reply:
Yes! good point!

Reply:
@teosarca actually no; is fine because  de.metas.i18n.Language.asLanguage(String) retunr null if language is  null